### ISSUE TITLE
"URL must be a string. Received undefined" during static export

### ISSUE THREAD

**Issue by sw-yx** [sveltejs/sapper#982]
**Describe the bug**
i'm running sapper export for my blog, and suddenly started receiving this new error: "URL must be a string. Received undefined".

![image](https://user-images.githubusercontent.com/6764957/68362381-5fc01e80-0127-11ea-80a7-04fabbf1a294.png)

I have no idea where this error message comes from, and so no idea how to fix it. still trying to figure out how to repro but figured i would log this issue first.

**To Reproduce**

not sure yet. if i know i'll update this

**Expected behavior**
the error message tells me how to fix it.

**Stacktraces**
 not yet

**Information about your Sapper Installation:**

- Sapper version (Please check you can reproduce the issue with the latest release!)

0.27.9

- Svelte version (Please check you can reproduce the issue with the latest release!)

3.12.1

- If it is an exported (npm run export) or dynamic application.

export

- Whether your application uses Webpack or Rollup

rollup

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Sapper entirely?

blocking me only right now.

**Comment by mustafa0x** [comment]
Could this be related to https://github.com/sveltejs/svelte/pull/3846 and https://github.com/sveltejs/svelte/pull/3862?

**Comment by vilarfg** [comment]
> Could this be related to [sveltejs/svelte#3846](https://github.com/sveltejs/svelte/pull/3846) and [sveltejs/svelte#3862](https://github.com/sveltejs/svelte/pull/3862)?

no, the only thing that #3846 or #3862 do differently is splitting (`String.prototype.split`) and encoding (`encodeURI`). Neither of those ever produce `undefined`.

Problem MUST be elsewhere :thinking:

**Comment by sw-yx** [comment]
whats really frustrating is i dont even know the source of the error message. no error stack, nothing in the  export stack i can see logs out this error message specifically. such a terrible dx! i bet once we find the source of thus issue we’ll be able to improve logging for lots of people not just me

**Comment by rixo** [comment]
I bet it's coming from here: https://github.com/sveltejs/sapper/blob/b1c4585c874fd156aaa36cc1d71abb6864faa279/src/cli.ts#L266

Otherwise you can try this trick: `console.error = console.log = console.trace` as the first line in your entry point.

**Comment by sw-yx** [comment]
@rixo that was exactly what i needed to do. it gave me the stack trace i needed to go identify and fix my problem:

![image](https://user-images.githubusercontent.com/6764957/68387207-18a15000-015e-11ea-9fca-d5a12ba210ab.png)


@pngwn @Conduitry  would you take a PR to log out the stack trace for these errors? i strongly believe we should not swallow the stack trace. just error messages alone aren't enough.

**Comment by pngwn** [comment]
@sw-yx sorry, just seen this. 

I agree we should not be swallowing the stack trace here. We definitely need to show that when something goes wrong like in this case.
